refactor(ast): `oxc_ast` crate re-export AST types from other crates

overlookmotel opened the pull request:
Some AST types are defined in crates outside `oxc_ast` crate. Re-export these types from `oxc_ast` crate for ease - so you don't need to remember where they're defined to use them.

overlookmotel commented:
* **#4773** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/4773?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @overlookmotel and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="11px" height="11px"/> <b>Graphite</b></a>
  <!-- Current dependencies on/for this PR: -->

overlookmotel commented:
@rzvxa @Dunqing I would have asked Boshen to review this, but he's off today. I can't see any downside to this, so I think safe to merge if one of you can confirm that you can't see a downside either.

Dunqing reviewed (approved):
Looks good!

rzvxa reviewed (approved):
Good call, I think we should move stuff out of `oxc_span` later on. Atom makes sense to be there but I can't say the same thing for the SourceType and its sub-types.

This PR helps if we want to do that.

overlookmotel committed:
refactor(ast): `oxc_ast` crate re-export AST types from other crates (#4773)

Some AST types are defined in crates outside `oxc_ast` crate. Re-export these types from `oxc_ast` crate for ease - so you don't need to remember where they're defined to use them.

overlookmotel commented:
Thanks both for reviewing.
